### ISSUE TITLE
[BUG] Custom `beanie.BsonBinary` type doesn't work with pydantic 2.6.0+

### ISSUE THREAD

**Issue by Luc1412** [BeanieODM/beanie#850]
**Describe the bug**
There has already been #651, which was solved by implementing `beanie.BsonBinary`. With the latest pydantic 2.6.0, this type breaks again.

**To Reproduce**
```python
class Test(Document):
    test: beanie.BsonBinary
```

**Expected behavior**
The type work as exspected.

**Actual behavior**
```py
pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for <class 'bson.binary.Binary'>. Set `arbitrary_types_allowed=True` in the model_config to ignore this error or implement `__get_pydantic_core_schema__` on your type to fully support it.

If you got this error by calling handler(<some type>) within `__get_pydantic_core_schema__` then you likely need to call `handler.generate_schema(<some type>)` since we do not call `__get_pydantic_core_schema__` on `<some type>` otherwise to avoid infinite recursion.

For further information visit https://errors.pydantic.dev/2.6/u/schema-for-unknown-type
```

**Comment by valentinoli** [comment]
It would be good if `BsonBinary` along with other custom types Beanie provides would be mentioned in the API Documentation. For instance at
https://beanie-odm.dev/api-documentation/custom-types/

**Comment by valentinoli** [comment]
This seems to have been fixed in Pydantic `v2.6.1`

[Pull Request](https://github.com/pydantic/pydantic/pull/8710)
[Release v2.6.1](https://github.com/pydantic/pydantic/releases/tag/v2.6.1)

**Comment by Luc1412** [comment]
Yeah, seems good now. Thanks for letting me know.
